subprojects/docs/src/docs/userguide/authoringMaintainableBuildScripts.adoc

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[authoring_maintainable_build_scripts]]
+== Authoring Maintainable Build Scripts
+
+Gradle build scripts combine the qualities of declarative build logic, expressiveness as well as flexibility and rigidity as needed. As a build script author it is easy to fall into the trap of striking the wrong balance or applying poor coding habits. This chapter describes best practices for writing your build script in a meaningful, yet flexible and efficient way.
+
+[NOTE]
+====
+The third-party link:https://github.com/nebula-plugins/gradle-lint-plugin[Gradle lint plugin] helps with enforcing a desired code style in a build script if you are looking for appropriate link:https://en.wikipedia.org/wiki/Lint_(software)[linting] automation.
+====
+
+[[sec:declare_task_group_and_description]]
+=== Declare the group and description for tasks
+
+Even new users to a build should to be able to find crucial information quickly and effortlessly. In Gradle you can declare a api:org.gradle.api.Task#setGroup(java.lang.String)[group] and a api:org.gradle.api.Task#setDescription(java.lang.String)[description] for any task of the build. The <<sec:listing_tasks,tasks report>> uses the assigned values to organize and render the task for easy discoverability. Assigning a group and description is most helpful for lifecycle tasks.
```

Comment:
> Assigning a group and description is most helpful for lifecycle tasks.

I wouldn't imagine users know what a "lifecycle" task is. I recommend we state that this should be done for any task that build users will invoke directly.

Reply:
done